### PR TITLE
Document healthchecks

### PR DESCRIPTION
**Pull request by rculbertson**

### CODE REVIEW

File: docs/user_manual.md [hunk 1]
```diff
@@ -139,6 +140,33 @@ A current best practice is to save all your job creation parameters in
...
+
+* Begin executing health checks, using exponential backoff from 1 second to 30 seconds.
+* If no health checks succeed after 5 minutes, kill the container. Helios will start it back up and try again.
+* If a health check succeeds
+  * Stop running health checks
+  * Register service with nameless (if job is configured to do so)
+  * Mark the job as RUNNING
+
+#### HTTP
+This health check will make an http request to the specified port and path, and consider a return
```
Comment (davidxia) [inline docs/user_manual.md]: capitalize "http" and remove the "," since it's not a compound sentence?

File: docs/user_manual.md [hunk 2]
```diff
@@ -139,6 +140,33 @@ A current best practice is to save all your job creation parameters in
...
+code of 2xx or 3xx as successful. HTTP health checks are specified in the form `port_name:path`,
+where `port_name` is the **name** of the exposed port (as set in the `--port` argument), and `path`
+is the path portion of the URL. Requests have a connect timeout of 500ms and a read timeout of 10s.
+
+    helios create --http-check http:health -p http=8080 -r foo/http=http
+
+#### TCP
+This health check is successful if it able to connect to the specified port. You must specify the
```
Comment (davidxia) [inline docs/user_manual.md]: "if it's"

File: docs/user_manual.md [hunk 3]
```diff
@@ -139,6 +140,33 @@ A current best practice is to save all your job creation parameters in
...
+
+When a job is started, Helios can optionally run a health check before registering your service with
+nameless. This prevents the service from receiving traffic before it is ready. After the container
+starts, Helios will execute the health check as follows.
+
+* Begin executing health checks, using exponential backoff from 1 second to 30 seconds.
+* If no health checks succeed after 5 minutes, kill the container. Helios will start it back up and try again.
+* If a health check succeeds
+  * Stop running health checks
+  * Register service with nameless (if job is configured to do so)
```
Comment (davidxia) [inline docs/user_manual.md]: "Register the service in service discovery" Won't always be nameless and this is public so external people won't know about nameless :)

File: docs/user_manual.md [hunk 4]
```diff
@@ -139,6 +140,33 @@ A current best practice is to save all your job creation parameters in
 version-controlled files in your project directory. This allows you to tie
 your helios job params to changes in your application code.
 
+### Health Checks
+
+When a job is started, Helios can optionally run a health check before registering your service with
+nameless. This prevents the service from receiving traffic before it is ready. After the container
```
Comment (davidxia) [inline docs/user_manual.md]: replace nameless with service discovery or service registrar. "before registering your service with whatever service registration plugin you provided."

File: docs/user_manual.md [hunk 5]
```diff
@@ -139,6 +140,33 @@ A current best practice is to save all your job creation parameters in
...
+starts, Helios will execute the health check as follows.
+
+* Begin executing health checks, using exponential backoff from 1 second to 30 seconds.
```
Comment (davidxia) [inline docs/user_manual.md]: For some reason, I found "using exponential backoff from 1 second to 30 seconds" confusing. Maybe "using exponential backoff that ranges randomly from 1 to 30 seconds."?

**Comment by rculbertson** [comment]
@dxia Good points. I addressed all of them in the second commit. Let me know what you think.

**Comment by rohansingh** [comment]
:+1:

**Comment by davidxia** [comment]
:+1:

**Pull request merged**